File: editor/en/[3000]Maximum Area of Longest Diagonal Rectangle.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[][]} dimensions
 * @return {number}
 */
var areaOfMaxDiagonal = function (dimensions) {
    let maxDiagonal = 0;
    let maxArea = 0;

    for (const [length, width] of dimensions) {
        const diagonal = Math.sqrt(length * length + width * width);
        const area = length * width;

        if (diagonal > maxDiagonal || diagonal === maxDiagonal && area > maxArea) {
            maxDiagonal = diagonal;
            maxArea = area;
        }
    }

    return maxArea;
};
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[3000]Maximum Area of Longest Diagonal Rectangle.java
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int areaOfMaxDiagonal(int[][] dimensions) {
        double maxDiagonal = 0;
        int maxArea = 0;

        for (int[] diemension : dimensions) {
            int length = diemension[0];
            int width = diemension[1];
            double diagonal = Math.sqrt(length * length + width * width);
            int area = length * width;

            if (diagonal > maxDiagonal || diagonal == maxDiagonal && area > maxArea) {
                maxDiagonal = diagonal;
                maxArea = area;
            }
        }

        return maxArea;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
